### ISSUE TITLE
Tableview cell and subviews being kept in memory due to strong retain cycle

### ISSUE THREAD

**Issue by renanstig** [scenee/FloatingPanel#564]
### Description

After adding a table view to be tracked by doing the following: 
`floatingPanel?.track(scrollView: pinListCardViewController.tableView)`, the cell and subviews related to the table view aren't being deallocated from the memory after the view is closed. 

I did some debugging and I found out that the pan gesture is keeping a strong reference to the class `FloatingPanel`, more specifically here: 

`panGestureRecognizer.floatingPanel = self`

The solution is quite simple:

In the class `FloatingPanelPanGestureRecognizer`, we should make a weak reference to the `FloatingPanel`, turning this:

`fileprivate var floatingPanel: FloatingPanel?` into this: `fileprivate weak var floatingPanel: FloatingPanel?`

### Expected behavior

The floating panel should be deallocated correctly after we close the view which it's being used.

### Actual behavior

You can see that the number of floating panel objects are increasing each time you open the view. 

<img width="952" alt="Screenshot 2022-09-06 at 11 03 27" src="https://user-images.githubusercontent.com/4225270/188594681-bc9680a2-bd74-4ee2-86cc-035638f14c94.png">

<img width="336" alt="Screenshot 2022-09-06 at 11 06 07" src="https://user-images.githubusercontent.com/4225270/188594818-83f2d20e-8af9-4c93-acfe-e4e8dd99a23c.png">

### Steps to reproduce

Set up the floating panel as suggested below:

```
       floatingPanel?.set(contentViewController: pinListCardViewController)
        floatingPanel?.track(scrollView: pinListCardViewController.tableView)
        floatingPanel?.surfaceView.cornerRadius = 16.0
```

After the view is shown, close the view and check the memory graph and you should see that the floating panel remains in the memory. 

**How do you display panel(s)?**

* Add as child view controllers

**How many panels do you displays?**

* 1 

### Environment

**Library version**

- FloatingPanel (1.4.0)

**Installation method**

* CocoaPods
 
**iOS version(s)**

iOS15

**Xcode version**

Version 13.4.1 (13F100)

**Comment by scenee** [comment]
This issue was already resolved since v1.5.0 😉

**Comment by renanstig** [comment]
@scenee , wow, you're right. I'm sorry for opening this issue.
